### PR TITLE
Participant profiles the switch

### PR DESCRIPTION
**Pull request by BroiSatse**
This PR switches all the association and code to use ParticipantProfile in place of EarlyCareerTeacherProfile and MentorProfile.

Note: The data has been already synced and migrated. New ECT and mentor profile automatically create ParticipantProfile records, so no data will be lost.

### CODE REVIEW

File: app/models/legacy_ect_profile.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
-class EarlyCareerTeacherProfile < ApplicationRecord
+class LegacyECTeacherProfile < ApplicationRecord
+  self.table_name = "early_career_teacher_profiles"
```
Comment (kicferk1) [inline app/models/legacy_ect_profile.rb:4]: When do we axe them?

File: app/models/profile_declaration.rb [hunk 2]
```diff
@@ -1,14 +1,8 @@
...
   belongs_to :participant_declaration
 
-  scope :uplift, lambda {
-                   joins(
-                     "left join early_career_teacher_profiles
-                      on declarable_type='EarlyCareerTeacherProfile'
-                      and declarable_id=early_career_teacher_profiles.id",
-                   ).merge(EarlyCareerTeacherProfile.uplift)
-                 }
+  scope :uplift, -> { where(participant_profile_id: ParticipantProfile::ECT.uplift.select(:id)) }
```
Comment (kicferk1) [inline app/models/profile_declaration.rb:7]: https://github.com/DFE-Digital/early-careers-framework/pull/635/files#r663994053 probably needs some conflict resolving with this, right?

File: app/models/legacy_ect_profile.rb (hunk 1 above)
Comment (BroiSatse) [reply app/models/legacy_ect_profile.rb:4]: Soon - I just keep them to ensure there are no data missing after the switch.

File: app/models/profile_declaration.rb (hunk 2 above)
Comment (BroiSatse) [reply app/models/profile_declaration.rb]: Already resolved :)

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**